### PR TITLE
Feature/policies manager must initiate ptu in case getting device consent from the user

### PR DESCRIPTION
**Pull request by istoimenova**
004_ATF_P_TC_PTU_DeviceConsent_from_User.lua

To run the script you need to replace
user_modules/shared_testcases/testCasesForPolicyTable.lua
user_modules/shared_testcases/testCasesForPolicyTableSnapshot.lua
user_modules/shared_testcases/testCasesForBuildingSDLPolicyFlag.lua
from #32. Common functions are under review.

@maria-mih please review the script according to template.

### CODE REVIEW

**Comment by istoimenova** [comment]
@dboltovskyi  and @OVikhrov, please review the script.

File: test_scripts/Polices/Policy_Table_Update/004_ATF_P_TC_PTU_DeviceConsent_from_User.lua [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      local RequestId = self.hmiConnection:SendRequest("SDL.GetUserFriendlyMessage", {language = "EN-US", messageCodes = {"DataConsent"}})
+
+      EXPECT_HMIRESPONSE( RequestId, {result = {code = 0, method = "SDL.GetUserFriendlyMessage"}})
+      :Do(function(_,data)
+          self.hmiConnection:SendNotification("SDL.OnAllowSDLFunctionality",
+            {allowed = true, source = "GUI", device = {id = config.deviceMAC, name = ServerAddress, isSDLAllowed = true}})
+
+          EXPECT_HMINOTIFICATION("SDL.OnStatusUpdate", { status = "UPDATE_NEEDED" })
+
+          testCasesForPolicyTableSnapshot:create_PTS(true, {
```
Comment (istoimenova) [inline test_scripts/Polices/Policy_Table_Update/004_ATF_P_TC_PTU_DeviceConsent_from_User.lua]: @istoimenova create_PTS should be renamed as verify_PTS

File: test_scripts/Polices/Policy_Table_Update/004_ATF_P_TC_PTU_DeviceConsent_from_User.lua [hunk 2]
```diff
@@ -0,0 +1,106 @@
+---------------------------------------------------------------------------------------------
```
Comment (maria-mih) [inline test_scripts/Polices/Policy_Table_Update/004_ATF_P_TC_PTU_DeviceConsent_from_User.lua]: Start Review

File: test_scripts/Polices/Policy_Table_Update/004_ATF_P_TC_PTU_DeviceConsent_from_User.lua [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+local testCasesForBuildingSDLPolicyFlag = require('user_modules/shared_testcases/testCasesForBuildingSDLPolicyFlag')
+local testCasesForPolicyTableSnapshot = require('user_modules/shared_testcases/testCasesForPolicyTableSnapshot')
+
+--[[ General Precondition before ATF start ]]
+-- commonFunctions:SDLForceStop()
+testCasesForBuildingSDLPolicyFlag:Update_PolicyFlag("EXTENDED_POLICY", "EXTERNAL_PROPRIETARY")
+testCasesForBuildingSDLPolicyFlag:CheckPolicyFlagAfterBuild("EXTENDED_POLICY","EXTERNAL_PROPRIETARY")
+commonSteps:DeleteLogsFileAndPolicyTable()
+
+--TODO: shall be removed when issue: "ATF does not stop HB timers by closing session and connection" is fixed
```
Comment (maria-mih) [inline test_scripts/Polices/Policy_Table_Update/004_ATF_P_TC_PTU_DeviceConsent_from_User.lua]: Please add your name here

**Comment by istoimenova** [comment]
@LuxoftAKutsan please merge

**Comment by istoimenova** [comment]
@LuxoftAKutsan please merge.

**Comment by LuxoftAKutsan** [comment]
@istoimenova unable ti merge due to a conflicts. Please resolve or close PR

**Pull request closed without merging**